tools/spectral/ipa/rulesets/functions/eachCustomMethodMustUseCamelCase.js

```diff
@@ -0,0 +1,28 @@
+import { getCustomMethod, isCustomMethod } from './utils/resourceEvaluation.js';
+import { hasException } from './utils/exceptions.js';
+import { casing } from '@stoplight/spectral-functions';
+
+const RULE_NAME = 'xgen-IPA-109-custom-method-must-use-camel-case';
+const ERROR_MESSAGE = 'The custom method must use camelCase format.';
+const ERROR_RESULT = [{ message: ERROR_MESSAGE }];
+
+export default (input, opts, { path }) => {
+  // Extract the path key (e.g., '/a/{exampleId}:method') from the JSONPath.
+  let pathKey = path[1];
+
+  if (!isCustomMethod(pathKey)) return;
+
+  if (hasException(input, RULE_NAME)) {
+    return;
+  }
+
+  let methodName = getCustomMethod(pathKey);
+  if (!methodName) {
+    return;
+  }
+
+  const isCamelCase = casing(methodName, { type: 'camel', disallowDigits: true });
+  if (isCamelCase !== undefined) {
+    return ERROR_RESULT;
+  }
```

Comment:
```suggestion
return casing(methodName, { type: 'camel', disallowDigits: true });
```

Reply:
It would return `undefined` if validated or something like `{... use camel case}` if not validated.  
Also, we disregard returning `ERROR_RESULT`.

Reply:
> If you want ot customize the error message, you can change the yaml here:
> message: '{{error}} http://go/ipa/109' -> message: 'Custom error message here. http://go/ipa/109'

I can send a custom message which includes the problematic method name

Reply:
Ah I see, yeah it would be nice to have `{methodName} must use camelCase`, let's go with original approach and add the evaluated string to the message 👍